shared/more/dumb-sheet.render.native.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import React, {Component} from 'react'
+import {Box, Text, Input, Button} from '../common-adapters'
+import {globalStyles} from '../styles/style-guide'
+import type {DumbMap} from './dumb'
+
+import CommonMap from '../common-adapters/dumb.native'
+// import LoginMap from '../login/dumb.desktop'
+// import SignupMap from '../login/signup/dumb.desktop'
+// import TrackerMap from '../tracker/dumb.desktop'
+// import PinentryMap from '../pinentry/dumb.desktop'
+
+import {dumbFilter, dumbIndex} from '../local-debug'
+import debounce from 'lodash/debounce'
+
+class Render extends Component<void, any, any> {
+  state: any;
+  _onFilterChange: (a: any) => void;
+
+  constructor (props: any) {
+    super(props)
+
+    this.state = {
+      filter: (dumbFilter && dumbFilter.toLowerCase()) || '',
+      filterShow: false,
+      index: dumbIndex || 0
+    }
+
+    this._onFilterChange = debounce(filter => {
+      this.setState({filter})
+    }, 300)
+  }
+
+  render () {
+    const componentMap: DumbMap = {
+      ...CommonMap
+      // ...LoginMap,
+      // ...SignupMap,
+      // ...TrackerMap,
+      // ...PinentryMap
+    }
+
+    const components = []
+
+    Object.keys(componentMap).forEach(key => {
+      if (this.state.filter && key.toLowerCase().indexOf(this.state.filter) === -1) {
+        return
+      }
+
+      const map = componentMap[key]
+      const Component = map.component
+      Object.keys(map.mocks).map((mockKey, idx) => {
+        const mock = {...map.mocks[mockKey]}
+        const parentProps = mock.parentProps
+        mock.parentProps = undefined
+
+        components.push(
+          <Box key={mockKey} style={styleBox}>
+            <Text type='Body' style={{marginBottom: 5}}>{mockKey}</Text>
+            <Box {...parentProps}>
+              <Component key={mockKey} {...mock} />
+            </Box>
+          </Box>
+        )
+      })
+    })
+
+    const ToShow = components[this.state.index % components.length]
+
+    return (
+      <Box>
+        {ToShow}
+        <Box style={stylesControls}>
+          <Text type='BodySmall'>{this.state.index}</Text>
```

Comment:
you get a little control thing, you can move through the components using < and >. the ... button shows the filter you can type in

![screen shot 2016-04-19 at 12 03 47 pm](https://cloud.githubusercontent.com/assets/516435/14646259/ce9251b6-0626-11e6-8721-4902fe0fc364.png)